packages/cozy-client/src/devtools/index.jsx

```diff
@@ -78,35 +81,101 @@ const DevToolsNavList = ({ selected, panels, onNav }) => {
   )
 }
 
+const useResizeStyles = makeStyles(theme => ({
+  root: {
+    height: 3,
+    width: '100%',
+    background: theme.palette.primary.main,
+    cursor: 'row-resize'
+  }
+}))
+
+const ResizeBar = ({ ...props }) => {
+  const theme = useTheme()
+  const classes = useResizeStyles(theme)
+  return <div className={classes.root} {...props} />
+}
+
 const DevToolsPanel = props => {
+  const { panels: userPanels, open } = props
   const panels = useMemo(() => {
-    if (props.panels) {
-      return [...defaultPanels, ...props.panels]
+    if (userPanels) {
+      return [...defaultPanels, ...userPanels]
     }
     return defaultPanels
-  }, [props.panels])
+  }, [userPanels])
   const [currentPanel, setCurrentPanel] = useState('queries')
+  const ref = useRef()
+
+  const [panelHeight, setPanelHeight] = useLocalState(
+    'cozydevtools__height',
+    DEFAULT_PANEL_HEIGHT
+  )
+  /**
+   * Copied/adapted from react-query
+   * https://github.com/tannerlinsley/react-query/blob/master/src/devtools/devtools.tsx
+   */
+  const handleDragStart = startEvent => {
+    if (startEvent.button !== 0) return // Only allow left click for drag
+
+    const node = ref.current
+    if (node === undefined) {
+      return
+    }
+
+    const dragInfo = {
+      originalHeight: node.getBoundingClientRect().height,
+      pageY: startEvent.pageY
+    }
+
+    const run = moveEvent => {
+      const delta = dragInfo.pageY - moveEvent.pageY
+      const newHeight = dragInfo.originalHeight + delta
+
+      setPanelHeight(newHeight)
+    }
+
+    const unsub = () => {
+      document.removeEventListener('mousemove', run)
+      document.removeEventListener('mouseUp', unsub)
+    }
+
+    document.addEventListener('mousemove', run)
+    document.addEventListener('mouseup', unsub)
+  }
```

Comment:
I had something similar in mind, but this code is much more smaller than I excepted! Well done

Reply:
All credits goes to the initial implementation inside react-query :)